src/modules/sensors/rc_update.cpp

```diff
@@ -369,10 +368,10 @@ RCUpdate::rc_poll(const ParameterHandles &parameter_handles)
 				 * will take us to the correct final index.
 				 */
 				manual.mode_slot = (((((_rc.channels[_parameters.rc_map_flightmode - 1] - slot_min) * num_slots) + slot_width_half) /
-						     (slot_max - slot_min)) + (1.0f / num_slots));
+						     (slot_max - slot_min)) + (1.0f / num_slots)) + 1;
 
-				if (manual.mode_slot >= num_slots) {
-					manual.mode_slot = num_slots - 1;
+				if (manual.mode_slot > num_slots) {
```

Comment:
I understand it correctly that this should actually never happen?

Reply:
Yes, this only happens if someone publishes a non-existing slot which never happens but you want to save yourself from switching into a random mode because of out of array range access.